desc/compute/utils.py

```diff
@@ -343,16 +343,25 @@ def get_transforms(keys, obj, grid, jitable=False, **kwargs):
         if hasattr(obj, c + "_basis"):  # regular stuff like R, Z, lambda etc.
             basis = getattr(obj, c + "_basis")
             # first check if we already have a transform with a compatible basis
-            for transform in transforms.values():
-                if basis.equiv(getattr(transform, "basis", None)):
-                    ders = np.unique(
-                        np.vstack([derivs[c], transform.derivatives]), axis=0
-                    ).astype(int)
-                    # don't build until we know all the derivs we need
-                    transform.change_derivatives(ders, build=False)
-                    c_transform = transform
-                    break
-            else:  # if we didn't exit the loop early
+            if not jitable:
```

Comment:
Are we sure this does what you want? It makes sense to assume that the compiler skips this if statement completely for jitable cases hence able to jit it without error, but I have never tried to put not jitable parts of the function inside an if statement. If it works this is cool

Reply:
Can you check `jax.jit(get_transforms)(..., jitable=True)` to see if it works in addition to test you added below? Don't add a new test but I have some doubts that this is actually jitable.

Reply:
This does work as intended, as long as jitable itself is static. We do a similar thing in the grid class